### PR TITLE
resource-recorder: update the docker compose file

### PR DESCRIPTION
**Pull request by iulianbarbu**
## Description of change

Added a docker-compose service for resource-recorder and `Makefile` tag env variable.

## How has this been tested? (if applicable)

Started Shuttle locally and it worked. Tested on unstable too.

### CODE REVIEW

File: resource-recorder/prepare.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+# Patch crates to be on same versions
+mkdir -p $CARGO_HOME
+touch $CARGO_HOME/config.toml
+if [[ $PROD != "true" ]]; then
+    echo '
+    [patch.crates-io]
+    shuttle-common = { path = "/usr/src/shuttle/common" }
+    shuttle-proto = { path = "/usr/src/shuttle/proto" }' > $CARGO_HOME/config.toml
+fi
```
Comment (chesedo) [inline resource-recorder/prepare.sh:21]: Why does resource recorder require patches? It does not compile anything like deployer does

Comment (iulianbarbu) [reply resource-recorder/prepare.sh:21]: True that. The workspace deps for common and proto are path based, so it's all good. Will change this.

File: .circleci/config.yml [hunk 2]
```diff
@@ -660,19 +660,19 @@ workflows:
               only: production
       - approve-push-unstable:
           type: approval
-          filters:
-            branches:
-              only: main
+          # filters:
+          #   branches:
+          #     only: main
```
Comment (chesedo) [inline .circleci/config.yml:665]: Just have to remove comments :smile:

**Review by chesedo — COMMENTED** [review]
Just have to undo comments

File: .circleci/config.yml (hunk 2 above)
Comment (iulianbarbu) [reply .circleci/config.yml]: Done.

File: cargo-shuttle/src/args.rs [hunk 3]
```diff
@@ -325,7 +325,7 @@ impl InitTemplateArg {
             Tide => "tide/hello-world",
             Tower => "tower/hello-world",
             Warp => "warp/hello-world",
-            None => "custom/none",
+            None => "custom-service/none",
```
Comment (chesedo) [inline cargo-shuttle/src/args.rs:328]: Not sure how the change relates to this PR?

**Review by chesedo — COMMENTED** [review]
PR just has an unexpected change

File: cargo-shuttle/src/args.rs (hunk 3 above)
Comment (iulianbarbu) [reply cargo-shuttle/src/args.rs:328]: It doesn't. It's a fix required for the tests to pass and a regression that came with custom-resources. I modified `shuttle-examples`, by changing the `custom` directory name to `custom-service`. The PR is merged on `shuttle-examples` main and will cause failure to initing a `none` example with `cargo-shuttle`.

Comment (chesedo) [reply cargo-shuttle/src/args.rs:328]: Ah, I see. This should ideally have been a separate PR

**Review by chesedo — APPROVED** [review]

**Pull request merged**